pkg/operator/controller/credentials-secret/controller_test.go

```diff
@@ -474,34 +578,145 @@ func testAzureExtDNSInstanceNoSecret() *operatorv1beta1.ExternalDNS {
 	return extDNS
 }
 
+func testAzureSrcSecret() *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testAzureSrcSecretName,
+			Namespace: testOperatorNamespace,
+		},
+		Data: map[string][]byte{
+			"azure.json": []byte("val1"),
+		},
+	}
+}
+
+func testAzureWrongSrcSecret() *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testAzureSrcSecretName,
+			Namespace: testOperatorNamespace,
+		},
+		Data: map[string][]byte{
+			"wrong-azure-config.json": []byte("val1"),
+		},
+	}
+}
+
+func testAzureTargetSecret() *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testTargetSecretName,
+			Namespace: testOperatorNamespace,
```

Comment:
The target secret has to be in the operand namespace, that's why you got `Added` event instead of no events.